src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+    private int maxSize;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+        maxSize = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == maxSize) {
+            increaseSize();
+        }
+        ++size;
+        array[size - 1] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index - 1 > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds");
+        }
+        if (size + 1 == maxSize) {
+            increaseSize();
+        }
+        System.arraycopy(array, index, array, index + 1, size++ - index);
+        array[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size + list.size() > DEFAULT_CAPACITY) {
+            throw new RuntimeException("IndexOut");
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index + 1 > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds");
+        }
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds");
+        }
```

Comment:
if this logic is the same in several methods - should we move it to private method?